controller/api/destination/endpoint_listener.go

```diff
@@ -29,15 +27,25 @@ type updateAddress struct {
 	pod     *corev1.Pod
 }
 
-func (ua updateAddress) String() string {
-	addrStr := addr.ProxyAddressToString(ua.address)
+func (ua *updateAddress) Address() string {
+	return addr.ProxyAddressToString(ua.address)
+}
+
+func (ua *updateAddress) Name() string {
 	if ua.pod == nil {
-		return fmt.Sprintf("{address:%v}", addrStr)
+		return ""
 	}
-	return fmt.Sprintf("{address:%v, pod:%s.%s}", addrStr, ua.pod.Namespace, ua.pod.Name)
+	return ua.pod.Name
 }
 
-func (ua updateAddress) clone() *updateAddress {
+func (ua *updateAddress) Namespace() string {
```

Comment:
curious about the motivation for breaking up `String()` into 3 methods? i think the goal of `String()` was to provide a unique, comparable representation of an `updateAddress` for logging and testing.

Reply:
It's only used in one place, though.

Reply:
It was used in `test_helper.go`, but now that same comparison requires calling 3 methods rather than 1.

It was also used implicitly in:
```
l.log.Debugf("Update(%+v, %+v)", add, remove)
```
... because it had previously satisfied the [Stringer interface](https://golang.org/pkg/fmt/#Stringer).